### PR TITLE
cpu-flexcpu: Added tracing for debug-flag Exec

### PR DESCRIPTION
**Pull request by animalvgamer**
Change-Id: Ic9d8f2e8ccb0bd2d5f493181f3d77d0059a0687e
Signed-off-by: Bradley Wang <radwang@ucdavis.edu>

### CODE REVIEW

File: src/cpu/flexcpu/inflight_inst.cc [hunk 1]
```diff
@@ -601,6 +606,8 @@ InflightInst::initiateMemRead(Addr addr, unsigned int size,
                               Request::Flags flags)
 {
     if (backingMemoryInterface) {
+        if (_traceData)
```
Comment (powerjg) [inline src/cpu/flexcpu/inflight_inst.cc]: One line or curly braces?

File: src/cpu/flexcpu/inflight_inst.cc [hunk 2]
```diff
@@ -56,6 +56,11 @@ InflightInst::InflightInst(ThreadContext* backing_context,
     staticInst(inst_ref);
 }
 
+InflightInst::~InflightInst()
+{
+    if (_traceData) delete _traceData;
```
Comment (powerjg) [inline src/cpu/flexcpu/inflight_inst.cc]: Why do you have this if you also have deleteTraceData?

Comment (animalvgamer) [reply src/cpu/flexcpu/inflight_inst.cc]: For things like faults that discard the InflightInst, I want to make sure it's cleared. deleteTraceData is so I can clear it earlier if we know we won't need it anymore I guess

Comment (powerjg) [reply src/cpu/flexcpu/inflight_inst.cc]: I see. I would pick one or the other. Can we just have the one in the destructor?

File: src/cpu/flexcpu/inflight_inst.cc (hunk 1 above)
Comment (animalvgamer) [reply src/cpu/flexcpu/inflight_inst.cc]: done

File: src/cpu/flexcpu/inflight_inst.cc (hunk 2 above)
Comment (animalvgamer) [reply src/cpu/flexcpu/inflight_inst.cc]: done

**Pull request merged**